app/controllers/settings_controller.rb

```diff
@@ -20,6 +20,12 @@ def lastfm_callback
     redirect_to :action => 'index'
   end
 
+  def lastfm_connect
+    key = Rockstar.lastfm_api_key
+    callback_url = url_for :action => 'lastfm_callback', :only_path => false
```

Comment:
Use the new Ruby 1.9 hash syntax.<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.